src/app/Http/Controllers/GetTasks.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace PhpTodoList\Http\Controllers;
+
+use Doctrine\DBAL\Connection;
+
+class GetTasks
+{
+    public function __construct(
+        protected Connection $dbConnection,
+    ) {
+    }
+
+    public function __invoke(): string
+    {
+      $queryBuilder = $this->dbConnection->createQueryBuilder();
+
+      $query = $queryBuilder
+        ->select('task', 'due_at')
+        ->from('tasks');
+        ;
+  
+        return $query;
```

Comment:
We want to return an array (ideally).

So do:-

```php
return $query->execute()->fetchAllAssociative();
```

This will execute your query and fetch all of the rows as an associative array. If you do `dd($query->execute()->fetchAllAssociative());` you'll see what it actually returns as an array.

Somewhere in the code I built for you, I'm JSON encoding whatever you return from your controllers so it's easier to work with on the frontend 👍